### ISSUE TITLE
[FR] Subsume Azure SQL as a plan of csb-azure-mssql-db

### ISSUE THREAD

**Issue by claassen** [pivotal/cloud-service-broker#110]
**Is your feature request related to a problem? Please describe.**
In order to make the transition from MASB Azure SQL to CSB and subsequent upgrades of service plans for subsumed instances match more closely with the current workflow that teams already use and reduce the amount of effort required modifying existing pipelines.

**Describe the solution you'd like**
Instead of a standalone service (azure-mssql-db-masb-subsume) being used to subsume an Azure SQL instance and then upgrades being performed with internal parameters, e.g.:

```
cf create-service csb-azure-mssql-db subsume <instance-name> -c '{ <params required for subsume> }'
cf update-service <instance-name> -c '{"service_objective":"new objective"}'
```

to have the subsume process simply be a plan of the csb-azure-mssql-db service, such that after subsuming is performed teams need only to change the service name in their existing pipeline scripts and updates to service plans can be executed using plan names in the usual way, e.g.:

```
cf update-service <instance-name> -p <plan-name>
```

The desired over all flow would look like:

```
# Initial subsume
cf create-service csb-azure-mssql-db subsume <instance-name> -c '{ <params required for subsume> }'

# Subsequent update of service plan
cf update-service <instance-name> -p <plan-name>
```

**Describe alternatives you've considered**
Using the existing solution with parameter updates described above.

**Additional Context**
We currently wrap the CF APIs directly with our own CLI which acts as a replacement for the CF CLI, such that `cf create-service` and `cf update-service` are the same command and it is automatically detected if an update needs to be performed rather than a create.

For example a team's pipeline would contain a line such as:

```
./provisioning-cli service create --name <instance name> --serviceName <service name> --planName <plan name>
```

where if the same command is run again with a different plan or different parameters we internally convert this into the equivalent of a `cf update-service` via a PUT call to the `/service_instances/<instanceGUID>` API.

This new proposed solution would allow teams to simply change the `<service name>` value in their pipelines and the workflow for updating services would be identical to the process for newly created (non-subsumed) instances.

**Priority**
High

**Priority Context**
The migration effort and communication of changes for the many teams that use Azure SQL that will need to migrate will be significantly reduced with this proposed approach.

**Platform** 
Azure

**Applicable Services**
csb-azure-mssql-db
azure-mssql-db-masb-subsume

**Comment by omerbensaadon** [comment]
Hey Michael! Totally agree that this is a better pattern for subsume going forward, thank you for the feedback.

@erniebilling just started a spike on this and we will respond with any info we have about implementation in this thread.

**Comment by omerbensaadon** [comment]
Shipped as apart of [v0.8.0-beta](https://github.com/pivotal/cloud-service-broker/releases/tag/v0.8.0-beta) TanzuNet release coming this week!
